### DISCUSSION TITLE
State of v3

### DISCUSSION THREAD

**Discussion by aolko** [fomantic/Fomantic-UI#2978] [General] [answered]
So, is v3 actually dead?

**Comment by lubber-de** [comment, marked as answer]
Yes and no. A v3 will eventually happen, but not as previously designed some years ago. No ETA yet. We keep maintaining v2

**Reply by aolko** [reply]
In that case please make components apis less painful to work with

**Reply by aolko** [reply]
> Yes and no. A v3 will eventually happen, but not as previously designed some years ago. No ETA yet. We keep maintaining v2

When? In another 5 years? Waiting since 2018.

**Reply by lubber-de** [reply]
Without knowing what will change, where is your need to upgrade? 😉 

Otherwise, you are welcome to contribute to the project. Basically v3 will be a complete rewrite, but tries to be backward compatible as good as possible , so projects should not need to also completely rewrite their apps as well

**Reply by lubber-de** [reply]
> In that case please make components apis less painful to work with

Any suggestions?

**Reply by aolko** [reply]
> > In that case please make components apis less painful to work with
> 
> Any suggestions?

can't tell from the get go, iirc the horrible one was select

**Reply by aolko** [reply]
Made [a thread with suggestions](https://github.com/fomantic/Fomantic-UI/discussions/2980)
